src/main/java/Main.java

```diff
@@ -187,4 +188,11 @@ public void pathTraversal2(HttpServletRequest request) throws IOException {
 
         }
     }
+    
+    void formatStringNoncompliant(final File file) {
+        final long length = file.length();
+        // Noncompliant: avoids using the correct format strings for their argument types.
+        final String s = String.format("File length is %s", length);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This code uses &#39;%s&#39; to format long: length (declared at line 193). This is a potential locale-sensitive handling issue. It might cause errors in the handling and processing of the statement at line: 195. Consider formatting this data with &#39;%d&#39; instead.